evaluator/evaluator/routers/dataset.py

```diff
@@ -1,22 +1,234 @@
+                        "mapping": {
+                            "id-column": item["mapping"]["id-column"],
+                            "question-column": item["mapping"]["question-column"],
+                            "context-column": item["mapping"]["context-column"],
+                            "answer-text-column": item["mapping"]["answer-text-column"],
+                        },
+                    }
+                else:
+                    return f" Undefined dataset_name"
+        else:
+            return "Dataset_name not exist on the datasets collection!"
 
-@router.get(
-    "",
-    response_model=List[str],
+    except ValueError as e:
+        msg = f"Dataset_name: {dataset_name} not found; Error: {e}"
+        logger.debug(msg)
+        raise HTTPException(404, msg)
+
+
+@router.put(
+    "/{dataset_name}/{skill_type}/{metric}",
+    status_code=200,
+)
+async def update_dataset(dataset_name: str, skill_type: str, metric: str):
+    logger.debug("put dataset")
+    metadata = get_dataset_metadata(dataset_name=dataset_name)
+
+    # check if the dataset_name exist on the collection mongodb
+    try:
+        dataset_result = mongo_client.client.evaluator.datasets.find(
+            {"dataset_name": dataset_name}
+        )
+    except ValueError as e:
+        msg = f" The dataset_name: {dataset_name} not be found; Error: {e}!"
+        logger.error(msg)
+        raise HTTPException(400, m)
+    for dataset_item in dataset_result:
+        if dataset_item["dataset_name"] is not None:
+            try:
+                myquery = {"dataset_name": dataset_item["dataset_name"]}
+                new_value = {"$set": {"skill-type": skill_type, "metric": metric}}
+                mongo_client.client.evaluator.datasets.update_one(myquery, new_value)
+
+                logger.debug(f"Dataset_name {dataset_name} is updated from mongodb!")
+                return f"Dataset_name: {dataset_item['dataset_name']} has be update"
+            except ValueError as e:
+                msg = (
+                    f" The dataset_name: {dataset_name} cannot be inserted; Error: {e}!"
+                )
+                logger.error(msg)
+                raise HTTPException(400, msg)
+    return dataset_name
+
+
+@router.delete(
+    "/{dataset_name}",
+    status_code=200,
 )
-async def get_datasets():
-    """Returns a list of supported data sets."""
-    datasets = [dataset.value for dataset in DataSet]
+async def delete_dataset(dataset_name: str):
+    logger.debug("Delete dataset_name on mongodb")
+
+    # check if dataset_name exist on the database
+    try:
+        result_db = mongo_client.client.evaluator.datasets.find(
+            {"dataset_name": dataset_name}
+        )
+        logger.debug(f"db_dataser_name = {result_db}")
+    except ValueError as e:
+        msg = f"Datasetr_name: {dataset_name} does not found; Error: {e}"
+        logger.error(msg)
+        raise HTTPException(404, msg)
+
+    for item in result_db:
```

Comment:
Wieso iteriest du hier über Ergebnisse aus der Datenbank? Es sollte ja nur max. ein Ergebnis geben in der Datenbank

Reply:
ich habe die iteration gelöscht. Ich hatte die Iteration gemacht, weil result_db ein mongodb cursor zurückgibt. Durch die Iteration kann ich dataset_name object zugreifen.

Reply:
Wieso nicht einfach find_one benutzen. Dann kriegst du auch nur eins zurück und keine Liste

Reply:
ich habe die Anpassung gemacht. Die Zeile nummer stimmen nicht mehr